File: testcases/x86_testcases/x86_output/test4/test4.c
#include <stdio.h>

int array_sum(int array[], int size) {
    int sum = 0;
    for (int i = 0; i < size; i++) {
        sum = sum + array[i];
    }
    return sum;
}

void main2() {
    int array[] = {1, 2, 3, 4, 7, 20, -4, 5};
    int length = 8;
    int sum = array_sum(array, length);
    printf("The array is given by");
    printf("\n");
    for (int i = 0; i < length; i++) {
        printf("%d", array[i]);
        printf("\n");
    }
    printf("The sum of array elements is ");
    printf("%d", sum);
    printf("\n");
    return;
}

int main() {
    main2();
    return 0;
}

File: testcases/x86_testcases/test4.py
def array_sum(array : list[int], size : int) -> int:
    sum : int = 0
    i : int = 0
    for i in range(size):
        sum = sum + array[i]
    return sum

def main():
    array : list[int] = [1, 2, 3, 4, 7, 20, -4, 5]
    length : int = 8
    sum : int = array_sum(array, length)
    print("The array is given by")
    i : int = 0
    for i in range(length):
        print(array[i])
    print("The sum of array elements is ", sum)
    return

if __name__ == "__main__":
    main()
